tests/integration_test/dns_resolver_test.go

```diff
@@ -31,56 +31,79 @@ func toFqdnWithRoot(dns string) string {
 
 func TestDnsResolver(t *testing.T) {
 	t.Parallel()
-	s := env.NewTestEnv(comp.TestDnsResolver, platform.EchoSidecar)
-
-	// Setup dns resolver.
-	backendHost := "dns-resolver-test-backend.example.com"
-	dnsRecords := map[string]string{
-		toFqdnWithRoot(backendHost): platform.GetLoopbackAddress(),
-	}
-	dnsResolver := comp.NewDnsResolver(s.Ports().DnsResolverPort, dnsRecords)
-	go func() {
-		if err := dnsResolver.ListenAndServe(); err != nil {
-			t.Fatalf("Failed to set udp listener %s\n", err.Error())
-		}
-	}()
-
-	// Check dns resolver's health.
-	dnsResolverAddress := fmt.Sprintf("%v:%v", platform.GetLoopbackAddress(), s.Ports().DnsResolverPort)
-	if err := comp.CheckDnsResolverHealth(dnsResolverAddress, backendHost, platform.GetLoopbackAddress()); err != nil {
-		t.Fatalf("DNS Resolver is not healthy: %v", err)
-	}
-
-	// Setup the whole test framework.
-	s.SetBackendAddress(fmt.Sprintf("http://%s:%v", backendHost, s.Ports().BackendServerPort))
-	args := []string{"--service_config_id=test-config-id",
-		"--rollout_strategy=fixed", "--healthz=/healthz", "--dns_resolver_addresses=" + dnsResolverAddress}
 
 	testCase := []struct {
 		desc            string
+		backendHost     string
 		isResolveFailed bool
 		wantResp        string
 		wantError       string
 	}{
 		{
-			desc:     "resolve domain name successfully",
-			wantResp: `{"message":"hello"}`,
+			desc:        "resolve non-FQDN domain name successfully",
+			backendHost: "dns-resolver-test-backend",
+			wantResp:    `{"message":"hello"}`,
+		},
+		{
+			desc:        "resolve FQDN domain name successfully",
+			backendHost: "dns-resolver-test-backend.example.com",
+			wantResp:    `{"message":"hello"}`,
+		},
+		{
+			desc:        "resolve workstation FQDN domain name successfully",
+			backendHost: "dns-resolver-test-backend.corp.google.com",
```

Comment:
confused,  will this fail when the test is run in prow, or in the hosts with other domain names?

If no,  it is just a FQDN,  it could be any FQDN, then is this case cover by the one above with "dns-resolver-test-backend.example.com"

Reply:
Same question for the next test

Reply:
Previously yet. Even when we used a FQDN like `dns-resolver-test-backend.corp.google.com`, it would work on our workstations but fail on prow. That is because the hostname in the prow job is `test-pods.svc.cluster.local`, so it would check for `dns-resolver-test-backend.corp.google.com.est-pods.svc.cluster.local`.

I put those cases here to make sure this test never fails on both our workstations and prow.